### DISCUSSION TITLE
local form of Test Replay?

### DISCUSSION THREAD

**Discussion by BloodyRain2k** [cypress-io/cypress#28272] [Questions and Help]
I was just wondering if there are plans, or even the chance to create plans, for having Test Replay work locally.

The cloud approach for it makes sense when the CI already runs in the cloud anyways, but for me running the project and Cypress locally, it just feels non-sensical to have to run it up into the cloud, while the entire thing happens literally in front of me, just so I can have the Test Replay functionality.

Obviously this begs the question why I'm not just using open mode, and the answer is simply that Cypress can't deal with SharePoint's flood of libraries.
Or rather that it can't stop snapshotting every single script and CSS file during testing, nor actually utilize the full RAM of this PC, and instead just commits die() once 4 GB are surpassed, regardless of 8 GB still being available...

It would already be a big help if I could just pipe the console output into the reporter, but after going as far as attempting to write my own, I realized that they run in so different context's that I can't figure out a way to get data from the test's console.log wrapper into said reporter...
I'm aware that there's a plugin for that, but as far as I've understood it would that require a different logger to work with, and I was hoping to avoid setting up yet another external part from scratch.

And all this because I can't just a record a Replay locally and go through it, if the runner happened to run into an error.
So yeah, just curious if there's any chance for this eventually becoming a thing.

**Comment by jennifer-shehane** [comment]
@BloodyRain2k We're focusing on performance issues and we have discussed updating our snapshotting logic in open mode with the same logic used for Test Replay. Right now there aren't plans to implement this, but your suggestion is noted.
